server/raft.go

```diff
@@ -3383,35 +3383,37 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 		n.updateLeadChange(false)
 	}
 
-	if (isNew && ae.pterm != n.pterm) || ae.pindex != n.pindex {
-		// Check if this is a lower index than what we were expecting.
-		if ae.pindex < n.pindex {
+	if ae.pterm != n.pterm || ae.pindex != n.pindex {
+		// Check if this is a lower or equal index than what we were expecting.
+		if ae.pindex <= n.pindex {
 			n.debug("AppendEntry detected pindex less than ours: %d:%d vs %d:%d", ae.pterm, ae.pindex, n.pterm, n.pindex)
```

Comment:
(This debug statement is now very slightly incorrect)

Reply:
This is how the debug statement was prior to me removing the equal check in the previous PR.

Have rephrased it to mention it can also be equal.